src/osd/PG.cc

```diff
@@ -759,7 +759,56 @@ void PG::MissingLoc::check_recovery_sources(const OSDMapRef& osdmap)
     }
   }
 }
-  
+
+void PG::MissingLoc::remove_stray_recovery_sources(pg_shard_t stray)
+{
+  set<pg_shard_t> now_stray;
+  for (set<pg_shard_t>::iterator p = missing_loc_sources.begin();
+       p != missing_loc_sources.end();
+       ) {
+    if (*p != stray) {
+      ++p;
+      continue;
+    }
+    ldout(pg->cct, 10) << __func__ << " source osd." << *p << " now stray" << dendl;
+    now_stray.insert(*p);
```

Comment:
@neha-ojha This `now_stray` is not defined in luminous... :-(

Reply:
now_stray was removed by "osd/MissingLoc.cc: do not rely on missing_loc_sources only" so the manual backport of 5906a57320f04f57a38eef9588bd16ac3fd4e55d isn't quite right.

Reply:
Thanks, @dzafman - fixed now!